internal/controller/controllers/agent_controller.go

```diff
@@ -137,15 +138,36 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 		}
 	} else { // agent is being deleted
 		if funk.ContainsString(agent.GetFinalizers(), AgentFinalizerName) {
+			if _, has_annotation := agent.GetAnnotations()[BMH_FINALIZER_NAME]; has_annotation {
+				// wait for BMH to be deleted
+				if foundBMH, err := r.bmhExists(ctx, agent); err != nil || foundBMH {
+					if err != nil {
+						log.WithError(err).Warnf("failed to determine if BMH exists for agent")
+					}
+					log.Info("waiting for BMH to be deleted")
+					return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, err
+				}
+				// delete spoke node
+				spokeClient, err := r.spokeKubeClient(ctx, agent.Spec.ClusterDeploymentName)
+				if err != nil {
+					log.WithError(err).Error("failed to create spoke client, node will not be removed")
+				} else {
+					nodeName := getAgentHostname(agent)
+					if err := spokeClient.DeleteNode(ctx, nodeName); err != nil {
```

Comment:
We should probably delete the BMH from the spoke as well.
Not sure if here or in BMAC

Reply:
Should we assume the spoke BMH is not being used?
In my tests the spoke BMHs are being used by a machine in a machineset.
Should we also try to clean that up?

Also isn't @slaviered making the spoke BMH effectively useless? Maybe we just shouldn't be creating them at all in the future?

Maybe this is better in a followup?

Reply:
What is the intention on having BMHs copied into the spokes? Maybe it is part of the installation, I dont know. But mainly there you have the BMCs information, that would be useful for spoke admin to interact with the BMC? 
Anyway, if the BMHs exists on the spoke, I think it should be deleted. If not, for example, I have experienced problems re-adding the node. So, you would want to delete one node, maybe some maintenance, hardware replacement, and after that.. you decide to re-install the host. If the BMH was not deleted, I have found the CSR is not automatically approved. In the same scenario, if I previously I deleted the BMH, you can re-install the host seamless.